backend/src/main/java/com/sim_backend/state/SimulatorStateMachine.java

```diff
@@ -0,0 +1,59 @@
+package com.sim_backend.state;
+
+import java.util.*;
+
+public class SimulatorStateMachine {
+    private SimulatorState currentState;
+    private Map<SimulatorState, Set<SimulatorState>> validation = new HashMap<>();
+    private List<StateIndicator> indicators = new ArrayList<>();
+
+    // Start from Poweroff
+    public SimulatorStateMachine() {
+        currentState = SimulatorState.PowerOff;
+        initValidation();
+        notifyIndicator();
+    }
+
+    // define which state can be used for a certain state
+    public void initValidation() {
+        validation.put(SimulatorState.PowerOff, Set.of(SimulatorState.BootingUp));
+        validation.put(SimulatorState.BootingUp, Set.of(SimulatorState.Available));
+        validation.put(SimulatorState.Available, Set.of(SimulatorState.PowerOff, SimulatorState.Reserved, SimulatorState.InUse));
+        validation.put(SimulatorState.Reserved, Set.of(SimulatorState.InUse, SimulatorState.Available));
+        validation.put(SimulatorState.InUse, Set.of(SimulatorState.Completed));
+        validation.put(SimulatorState.Completed, Set.of(SimulatorState.Available));
+    }
+
+    // State Transition
+    public boolean transition(SimulatorState userInput){
+        if(validation.getOrDefault(currentState, null).contains(userInput)){
+            currentState = userInput;
+            notifyIndicator();
+            return true;
+        }
+        else{
+            System.out.println("You can't change it to" + userInput + "When it's " + currentState);
+            return false;
+        }
+    }
+
+    // Popup Message State Change
+    private void notifyIndicator() {
```

Comment:
The logic implemented previously for the `notifyIndicator` function and the `StateIndicator.java` file was spot on! 

This change breaks the observer pattern. What you implemented before was correct and worked well.

See previous commit for original logic in that `notifyIndicator` function and the `StateIndicator.java` file.
https://github.com/pglob/ocpp-charger-sim/pull/13/commits/c4fa23dff81bb0ff073102c676189c08fb6e84b9